src/libstd/sys/windows/path.rs

```diff
@@ -50,35 +47,35 @@ pub fn parse_prefix(path: &OsStr) -> Option<Prefix<'_>> {
                     return Some(VerbatimUNC(server, share));
                 } else {
                     // \\?\path
-                    let idx = path.iter().position(|&b| b == b'\\');
-                    if idx == Some(2) && path[1] == b':' {
-                        let c = path[0];
-                        if is_valid_drive_letter(c) {
-                            // \\?\C:\ path
+                    match path {
+                        // \\?\C:\path
+                        [c, b':', b'\\', ..] if is_valid_drive_letter(*c) => {
                             return Some(VerbatimDisk(c.to_ascii_uppercase()));
                         }
+                        // \\?\cat_pics
+                        _ => {
+                            let idx = path.iter().position(|&b| b == b'\\').unwrap_or(path.len());
+                            let slice = path.get_unchecked(..idx);
```

Comment:
This is calling an `unsafe` function, but the safety comment at the top of the huge `unsafe` block does not mention that. This should either get its own safety comment or just not use `unsafe` if it doesn't seem worth it.